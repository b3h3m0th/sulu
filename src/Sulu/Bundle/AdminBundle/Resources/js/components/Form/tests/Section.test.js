// @flow
import React from 'react';
import {render} from '@testing-library/react';
import Section from '../Section';

test('Render section with given colSpan', () => {
    const {container} = render(
        <Section label="Test">
            <p>Test</p>
        </Section>
    );
    expect(container).toMatchSnapshot();
});

test('Render section without label', () => {
    const {container} = render(
        <Section colSpan={8}>
            <div>Test</div>
        </Section>
    );
    expect(container).toMatchSnapshot();
});

test('Render section without label but with divider', () => {
    const {container} = render(
        <Section>
            <p>Test</p>
        </Section>
    );
    expect(container).toMatchSnapshot();
});
