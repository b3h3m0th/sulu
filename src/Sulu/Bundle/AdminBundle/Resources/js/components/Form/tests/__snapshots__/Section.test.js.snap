// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render section with given colSpan 1`] = `
<div>
  <div
    class="section gridSection colSpan colSpan-12 space-before-0 space-after-0"
  >
    <div
      class="item dividerContainer colSpan colSpan-12 space-before-0 space-after-0"
    >
      <div
        class="divider"
      >
        Test
      </div>
    </div>
    <p>
      Test
    </p>
  </div>
</div>
`;

exports[`Render section without label 1`] = `
<div>
  <div
    class="section gridSection colSpan colSpan-8 space-before-0 space-after-0"
  >
    <div>
      Test
    </div>
  </div>
</div>
`;

exports[`Render section without label but with divider 1`] = `
<div>
  <div
    class="section gridSection colSpan colSpan-12 space-before-0 space-after-0"
  >
    <div
      class="item dividerContainer colSpan colSpan-12 space-before-0 space-after-0"
    >
      <div
        class="divider"
      />
    </div>
    <p>
      Test
    </p>
  </div>
</div>
`;
